.github/workflows/test-unreleased.yml

```diff
@@ -0,0 +1,60 @@
+name: Integration with Unreleased OpenSearch
```

Comment:
May be add the workflow in a different PR?

Reply:
I added it in the same PR since the breaking changes need to be tested. WDYT? I can add the workflow in a separate PR and include only 1.x branch on it, then in this PR I can add the 2.x, 2.0 and main branch testing.

Reply:
Haha its ok. 
Its easier to split pieces when they are independent, we can do it next time :)